drivers/vultr/vultr.go

```diff
@@ -0,0 +1,517 @@
+package vultr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/ChimeraCoder/tokenbucket"
+	vultr "github.com/JamesClonk/vultr/lib"
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/log"
+	"github.com/docker/machine/ssh"
+	"github.com/docker/machine/state"
+)
+
+type Driver struct {
+	APIKey            string
+	MachineID         string
+	MachineName       string
+	IPAddress         string
+	PrivateIP         string
+	OSID              int
+	RegionID          int
+	PlanID            int
+	SSHKeyID          string
+	SSHUser           string
+	SSHPort           int
+	IPv6              bool
+	Backups           bool
+	PrivateNetworking bool
+	CaCertPath        string
+	PrivateKeyPath    string
+	DriverKeyPath     string
+	SwarmMaster       bool
+	SwarmHost         string
+	SwarmDiscovery    string
+	storePath         string
+	scriptID          int
+	bucket            *tokenbucket.Bucket
+}
+
+func init() {
+	drivers.Register("vultr", &drivers.RegisteredDriver{
+		New:            NewDriver,
+		GetCreateFlags: GetCreateFlags,
+	})
+}
+
+// GetCreateFlags registers the flags this driver adds to "docker hosts create"
+func GetCreateFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			EnvVar: "VULTR_API_KEY",
+			Name:   "vultr-api-key",
+			Usage:  "Vultr API key",
+		},
+		cli.IntFlag{
+			EnvVar: "VULTR_OS",
+			Name:   "vultr-os-id",
+			Usage:  "Vultr operating system ID. Default: 159 (RancherOS)",
+			Value:  159,
```

Comment:
@hairyhenderson I have changed the default OS back to Ubuntu. Since the iPXE deployment of RancherOS is the better choice for Vultr (deploying Ubuntu takes up to 5 minutes) i have added instructions to the docs concerning the possibility to use RancherOS as base OS.

/cc @imikushin